Tests

leviLThomas opened the pull request:
## Description

This PR accomplishes the following:

- 100% collection queries coverage
- 97% memento queries coverage
- 100% hdbscan coverage
- 100% clustering queries coverage

![image](https://github.com/user-attachments/assets/a3944b8f-4005-46bd-afd5-44e84e13b3a6)

leviLThomas committed:
Omit Directories Outside Test Plan Goals

Coverage report includes only files in the server/services directory as
specified in the project test plan.

leviLThomas committed:
Collection Query Tests

89% Collection query coverage. Provided by Claude 3.7 Sonnet.

leviLThomas committed:
test delete memento

leviLThomas committed:
Only collect labels if they are not none

leviLThomas committed:
Memento Tests and Fixtures

Coverage of filter queries and image label queries

leviLThomas committed:
Fix filter date order validator

leviLThomas committed:
Fixing label calls

leviLThomas committed:
Fixing bounding box filter calls

leviLThomas committed:
Fix all_bounds_values validator

Previously used `any()` and `all()` on values, which treated 0.0 as
falsy. This caused valid coordinates to be misidentified as missing. Now
explicitly checks for `None` to ensure correct validation.

leviLThomas committed:
Rest of collection query tests

leviLThomas committed:
Linting

leviLThomas committed:
Merge branch 'main' into levi-testing

leviLThomas committed:
test create_user_info

leviLThomas committed:
test clustering queries

leviLThomas committed:
test hdbscan clustering

leviLThomas committed:
ignore warning from hdbscan library

leviLThomas committed:
Linting

leviLThomas commented:
Only 97% coverage on queries/memento.py because unable to reach early return. I suspect it has something to do with how the rpc return value is mocked, but I was unable to fix it.

owencooke committed:
100% memento coverage

owencooke commented:
> Only 97% coverage on queries/memento.py because unable to reach early return. I suspect it has something to do with how the rpc return value is mocked, but I was unable to fix it.

Fixed. Issue for the test was that the min lat/long == 0, which meant in `get_mementos` the `if all([filter_query.min_lat,...` was not being executed. So the RPC nor early return were hit

owencooke reviewed (approved):
✅ 94% total. So close!

leviLThomas committed:
AI Prompts

leviLThomas committed:
Create collection flow

Need to consider running multiple create memento's before

leviLThomas committed:
View collection flow

leviLThomas committed:
removal flows

leviLThomas committed:
Modularize create memento

top level create-memento.yml is a wrapper for create-test-memento util.

leviLThomas committed:
Create memento before collection

leviLThomas committed:
GIS Tests

leviLThomas committed:
Image Processor Tests

leviLThomas committed:
Test text_manager

leviLThomas committed:
Test Image Class

Tests fails with ValueError

leviLThomas committed:
Create Collection flow with expected setup

leviLThomas committed:
Edit Collection Flow with required setup

leviLThomas committed:
View Collection Flow with setup

leviLThomas committed:
Delete Collection Flow

leviLThomas committed:
Move Collection Flows to Subfolder

Collection Flows require manual setup. They should be executed manually
after required setup.

leviLThomas committed:
filter test ids

leviLThomas committed:
filter master flow

leviLThomas committed:
filter sub flows

leviLThomas committed:
fix #328

Return empty dict when too few mementos

leviLThomas committed:
fix #325

Revert to original image label filter strategy. Requires an additional
supabase api call.

leviLThomas committed:
Merge branch '325-image-label-filter-excludes-images-on-a-memento' into maestro-tests

leviLThomas committed:
Merge branch '328-recommend-collection-when-not-enough-mementos' into maestro-tests

leviLThomas committed:
patch input-text

https://github.com/mobile-dev-inc/Maestro/issues/395#issuecomment-2750427472

patch

linting

leviLThomas committed:
reduce filter flow mementos

leviLThomas committed:
update for input-text

update for input text

leviLThomas committed:
revert some changes

leviLThomas committed:
Merge branch 'maestro-tests' into levi-testing

leviLThomas committed:
use patch in filter-location

leviLThomas committed:
revert from patch

majamal014 committed:
100% coverage

majamal014 committed:
Merge branch 'levi-testing' of github.com:owencooke/memento into levi-testing

majamal014 committed:
fix cluster tests; still problem with test_get_mementos_image_label_filter

leviLThomas committed:
fix image_label test

New get_mementos query has a separate query for getting memento ids with
associated images that have image_label. Had to mock execution chain and
add side effect to supabase execute to get different responses.

leviLThomas committed:
Type annotations and other ruff lint

leviLThomas committed:
fix create-collection flow

- Uses same setup data as filter flow
- always selects edmonton memento
- date assertion broken by sticky datepicker

leviLThomas committed:
modify collection form for edit collection

leviLThomas committed:
fix edit-collection flow

- uses same setup data as filter
- correctly selects mementos

problems:
- not selected autocomplete result

leviLThomas committed:
add covered frs to create-collection

leviLThomas committed:
modularize individual collection flows

leviLThomas committed:
collection flow runner

owencooke committed:
Merge remote-tracking branch 'origin/main' into levi-testing

owencooke committed:
Merge branch 'levi-testing' of github.com:owencooke/memento into levi-testing

owencooke committed:
fix logs

owencooke committed:
replace console errors with logs

leviLThomas committed:
jest coverage report only show libs

owencooke committed:
remove header unit test

owencooke committed:
bull create passes

owencooke committed:
100% branch for photos

leviLThomas committed:
Use env in create-collection

owencooke committed:
fix edit memento bug

owencooke committed:
Merge branch 'levi-testing' of github.com:owencooke/memento into levi-testing

owencooke committed:
these e2e pass

owencooke committed:
python branch cov

owencooke committed:
map/collage should pass (emulator)

owencooke committed:
RISKY: potentially fixed location input

owencooke committed:
add instructions for tas

leviLThomas committed:
like image label not in_

leviLThomas committed:
Using shorter location partial in filter

leviLThomas committed:
No scroll needed on filter location

owencooke committed:
add scroll fix

owencooke committed:
fix map view

owencooke committed:
push

owencooke reviewed (approved):
Last LGTM 👍 😢
